pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -676,6 +580,226 @@ func closeFeed(t testing.TB, f cdctest.TestFeed) {
+// of a test running as the system tenant or a secondary tenant, while still
+// having access to system interfaces for operations that require them such as
+// desctestutils.TestingGetPublicTableDescriptor
+type TestServer struct {
+	DB           *gosql.DB
+	Server       serverutils.TestTenantInterface
+	SystemDB     *gosql.DB
+	SystemServer serverutils.TestServerInterface
+	Codec        keys.SQLCodec
+	TestingKnobs base.TestingKnobs
+}
+
+func makeSystemServer(
+	t *testing.T, opts ...feedTestOption,
+) (testServer TestServer, cleanup func()) {
+	options := makeOptions(opts...)
+	return makeSystemServerWithOptions(t, options)
+}
+
+var _ = makeSystemServer // silence unused warning
+
+func makeSystemServerWithOptions(
+	t *testing.T, options feedTestOptions,
+) (testServer TestServer, cleanup func()) {
+	systemServer, systemDB, clusterCleanup := startTestFullServer(t, options)
+	return TestServer{
+			DB:           systemDB,
+			Server:       systemServer,
+			SystemDB:     systemDB,
+			SystemServer: systemServer,
+			TestingKnobs: systemServer.(*server.TestServer).Cfg.TestingKnobs,
+			Codec:        keys.SystemSQLCodec,
+		}, func() {
+			clusterCleanup()
+		}
+}
+
+func makeTenantServer(
+	t *testing.T, opts ...feedTestOption,
+) (testServer TestServer, cleanup func()) {
+	options := makeOptions(opts...)
+	return makeTenantServerWithOptions(t, options)
+}
+func makeTenantServerWithOptions(
+	t *testing.T, options feedTestOptions,
+) (testServer TestServer, cleanup func()) {
+	systemServer, systemDB, clusterCleanup := startTestFullServer(t, options)
+	tenantID, tenantServer, tenantDB, tenantCleanup := startTestTenant(t, systemServer, options)
+
+	return TestServer{
+			DB:           tenantDB,
+			Server:       tenantServer,
+			SystemDB:     systemDB,
+			SystemServer: systemServer,
+			TestingKnobs: tenantServer.(*server.TestTenant).Cfg.TestingKnobs,
+			Codec:        keys.MakeSQLCodec(tenantID),
+		}, func() {
+			tenantCleanup()
+			clusterCleanup()
+		}
+}
+
+func makeServer(t *testing.T, opts ...feedTestOption) (testServer TestServer, cleanup func()) {
+	options := makeOptions(opts...)
+	return makeServerWithOptions(t, options)
+}
+
+func makeServerWithOptions(
+	t *testing.T, options feedTestOptions,
+) (testServer TestServer, cleanup func()) {
+	if options.useTenant {
+		t.Logf("making server as secondary tenant")
+		return makeTenantServerWithOptions(t, options)
+	}
+	t.Logf("making server as system tenant")
+	return makeSystemServerWithOptions(t, options)
+}
+
+func randomSinkType(opts ...feedTestOption) string {
+	options := makeOptions(opts...)
+	return randomSinkTypeWithOptions(options)
+}
+
+func randomSinkTypeWithOptions(options feedTestOptions) string {
+	sinkWeights := map[string]int{
+		"kafka":        2, // run kafka a bit more often
+		"enterprise":   1,
+		"webhook":      1,
+		"pubsub":       1,
+		"sinkless":     0, // default to no sinkless to avoid surprises
```

Comment:
I don't know why we would want to do this? What kind of surprises?

Reply:
I think we should keep sinkless -- if we have bugs in there, we should fix them.

Reply:
Well, by "bug" here I mean a test "bug" where the test shouldn't run on sinkless since it mentions job related information.  The fix would be adding `feedTestOmitSinks("sinkless")`

Reply:
I think the default should include all sinks.  

I think we should have all of the flavors take optional ... options.  So, if you have a test that wants to run on all but sinkless, then let's add
  cdcTest(...., allButSinkless)
But let's not pre-emptively add things - -i.e if we have a need for `allButSinkless` option -- great, let's add it.
We can also have more generic options, like:
 cdcTest(...., onlySinks('sinkless', 'cloudstorage')
Or we can add
  cdcTest(...., excludeSinks('blah'))
We can also add "useful" shortcuts, like enterpriseSinks() -- basically all but sinkless.

But by default, let's run all sinks.  If the test cannot run on a particular sink, they can express this desire by passing
appropriate option.

Reply:
All sinks is now default, with `feedTestEnterpriseSinks` as the "no-sinkless" option